pkg/transform/engine/engine_javascript.go

```diff
@@ -331,10 +334,39 @@ func mkJSEngineInput(e *JSEngine, message *models.Message, interState any) (*eng
 		HTTPHeaders: message.HTTPHeaders,
 	}
 
+	if e.SpModeV2 {
+		var cInput map[string]any
+		// Trying to parse as TSV
+		cInput, err := parseAsTSV(interState, message)
```

Comment:
There's something I didn't consider here which needs some thought.

We want to process the TSV first, because we don't want to add latency to the main production use case.

However there's an unintended consequence of that: When we parse a tsv, we set the parsedEvent format into `interState`, also for performance reasons.

So, if a previous transformation parsed the TSV, then performed a transformation on the data, its output wouldn't be used - instead we'd end up using the original TSV because of this line. But in that scenario, we should use the transformed data!

I think this might be an interesting one to discuss together as a team, let's find space to do so in today's meeting.

Reply:
Just to record the outcome: for now, we can set `interState` to `nil` when we call this function, and we avoid the problem.

Longer term, we should look at this functionality and find a better pattern.